### PR TITLE
[PUB-2866] Tabs refactoring

### PR DESCRIPTION
**Pull request by anafilipadealmeida**
## Description
- Use react-router to select tabs and navigate instead of redux
- Simplify tabs validation utils. Now we generate an array of what tabs each profile has.
- Fix bug preventing from navigating to a subtab
- Add accessibility improvements to navbar
- Move TabTag to shared-components and rename it as NavTag, as it's currently used by multiple navs (profile sidebar, profile nav)

## Context & Notes
https://buffer.atlassian.net/browse/PUB-2866
In another PR: remove publish-tabs package and `<Tab>` and `<Tabs>` shared-components. Also take care of the tabs in the DisabledQueue.

## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. 
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

**Review by hamstu — APPROVED** [review]
Thanks so much for working on this @anafilipadealmeida! Such a great step towards refactoring the routing. 🙏 💯

File: packages/tabs/middleware.js [hunk 1]
```diff
@@ -9,28 +9,28 @@ export default ({ getState, dispatch }) => next => action => {
...
-        action.profileId !== getState().profileId
-      ) {
-        dispatch(
-          profileTabPages.goTo({
-            profileId: action.profileId,
-            tabId: action.tabId,
-          })
-        );
-      }
+      // if (
```
[the remark this answers is not in the document]
Comment (anafilipadealmeida) [reply packages/tabs/middleware.js:12]: I'm thinking about cutting the scope of this a bit, and delete the entire package on a separated PR (as there are a few places still using small things from the tabs package)

File: packages/drafts/middleware.js [hunk 2]
```diff
@@ -21,30 +21,35 @@ const getTrackingData = ({ post = {}, channel = {} }) => ({
...
-            name: 'draftPosts',
-            args: {
-              profileId: action.profileId,
-              isFetchingMore: false,
-              needsApproval,
-              clear: true,
-            },
-          })
-        );
+    case LOCATION_CHANGE:
```
[the remark this answers is not in the document]
Comment (anafilipadealmeida) [reply packages/drafts/middleware.js]: Great idea, I've replaced it 😄

File: packages/shared-components/NavLink/index.jsx [hunk 3]
```diff
@@ -66,6 +67,7 @@ const NavLink = ({
         $secondary={secondary}
         to={to}
         href={href}
+        aria-haspopup={hasSubMenu ? true : undefined}
```
Comment (albennett) [inline packages/shared-components/NavLink/index.jsx:70]: 😍

**Review by albennett — APPROVED** [review]
Great job, @anafilipadealmeida 🙌 I did some manual testing, too, and everything looked great. 🚀 I look forward to all the -lines in the next related PR.

**Pull request merged**